include/fiction/algorithms/physical_design/apply_gate_library.hpp

```diff
@@ -207,6 +209,30 @@ class apply_gate_library_impl
             }
         }
     }
+    /**
+     * Computes the (inclusively) bounding coordinate for a cell-level layout that is derived from the dimensions of the
+     * given gate-level layout, while respecting tiling geometry in which even and odd rows/columns do not line up.
+     *
+     * @param gate_lyt Gate-level layout of which the dimensions are read.
+     * @return Aspect ratio for a cell-level layout that corresponds to the dimensions of the given gate-level layout.
+     */
+    static aspect_ratio<CellLyt> get_aspect_ratio_for_cell_level_layout(const GateLyt& gate_lyt) noexcept
+    {
+        const std::function<cell<CellLyt>(GateLyt, tile<GateLyt>, cell<CellLyt>)> rel_to_abs_cell_pos =
+            relative_to_absolute_cell_position<GateLibrary::gate_x_size(), GateLibrary::gate_y_size(), GateLyt,
+                                               CellLyt>;
```

Comment:
It might be possible to write this without an assignment to a function object (untested though)
```suggestion
        using rel_to_abs_cell_pos = relative_to_absolute_cell_position<GateLibrary::gate_x_size(), GateLibrary::gate_y_size(), GateLyt, CellLyt>;
```